src/string.cr

```diff
@@ -2200,6 +2200,45 @@ class String
     match_result.try &.begin(0)
   end
 
+  # Searches sep or pattern (regexp) in the string,
+  # and returns the part before it, the match, and the part after it.
+  # If it is not found, returns two empty strings and str.
+  #
+  # ```
+  # "hello".rpartition("l") # => {"he", "l", "lo"}
+  # "hello".rpartition("x") # => {"hello", "", ""}
+  # ```
+  def partition(search : (Char | String)) : Tuple(String, String, String)
+    pre, mid, post = {"", "", ""}
```

Comment:
I think this is much simpler:

```cr
pre = mid = post = ""
```

But in fact maybe you could assign these variables in each case, or maybe just return a different tuple in every case branch.